src/scheduled_tasks.rs

```diff
@@ -228,19 +228,27 @@ async fn process_post_aggregates_ranks_in_batches(conn: &mut AsyncPgConnection)
   while let Some(previous_batch_last_published) = previous_batch_result {
     let result = sql_query(
       r#"WITH batch AS (SELECT pa.post_id
-               FROM post_aggregates pa
-               WHERE pa.published > $1
-               AND (pa.hot_rank != 0 OR pa.hot_rank_active != 0)
-               ORDER BY pa.published
-               LIMIT $2
-               FOR UPDATE SKIP LOCKED)
-         UPDATE post_aggregates pa
-           SET hot_rank = r.hot_rank(pa.score, pa.published),
-           hot_rank_active = r.hot_rank(pa.score, pa.newest_comment_time_necro),
-           scaled_rank = r.scaled_rank(pa.score, pa.published, ca.users_active_month)
-         FROM batch, community_aggregates ca
-         WHERE pa.post_id = batch.post_id and pa.community_id = ca.community_id RETURNING pa.published;
-    "#,
+           FROM post_aggregates pa
+           WHERE pa.published > $1
+           AND (pa.hot_rank != 0 OR pa.hot_rank_active != 0)
+           ORDER BY pa.published
+           LIMIT $2
+          FOR UPDATE SKIP LOCKED),
+      community_interactions AS (
+          SELECT community_id,
+                 SUM(comments + upvotes + downvotes) as total_interactions
+          FROM post_aggregates
+          WHERE published >= date_trunc('month', CURRENT_TIMESTAMP - interval '1 month')
+          GROUP BY community_id)
```

Comment:
Add an `interactions_month` column to the `community_aggregates` table using a migration, and use it here just like how `users_active_month` was used, which will be more efficient. In utils.sql, add a variant of the `r.community_aggregates_activity` function that does the sum, and use it to update the new column in the `active_counts` function in this file.

To prevent a compile error, you also need to add the new column in crates/db_schema/src/aggregates/structs.rs at the end of the struct, and it should probably have the `#[serde(skip)]` attribute.